src/repositories/projectAddressRepository.ts

```diff
@@ -0,0 +1,91 @@
+import { ProjectAddress } from '../entities/projectAddress';
+import { Project } from '../entities/project';
+import { User } from '../entities/user';
+
+export const getPurpleListAddresses = async (): Promise<
+  { projectAddress: string }[]
+> => {
+  // addresses that are related to verified projects
+  const addresses = await ProjectAddress.query(
+    `
+          SELECT "projectId", LOWER(address) as "projectAddress"
+          FROM project_address
+          JOIN project
+          on project.id="projectId" and "verified"=true
+      `,
+  );
+  return addresses;
+};
+
+export const isWalletAddressInPurpleList = async (
+  address: string,
+): Promise<boolean> => {
+  const projectAddress = await ProjectAddress.query(
+    `
+          SELECT "projectId", LOWER(address) as "projectAddress"
+          FROM project_address
+          JOIN project
+          on project.id="projectId" and "verified"=true
+          where address='${address}'
```

Comment:
We have an sql injection problem here. Specially since the query ```walletAddressIsPurpleListed``` accepts the address directly. We have to escape it!!

Reply:
Fixed.!

Reply:
Thanks so much